scripts/common/kubernetes.sh

```diff
@@ -784,7 +784,15 @@ function kubeadm_cluster_status() {
 }
 
 function check_network() {
-	logStep "Checking cluster networking"
+	  logStep "Checking cluster networking"
+
+    if [ -n "$WEAVE_VERSION" ]; then
+        log "Checking if weave-net binary can be found in the path /opt/cni/bin/"
+        if ! ls -la /opt/cni/bin/ | grep weave-net; then
+            logWarn "Unable to find weave-net binary. Deleting weave-net pod"
+            kubectl delete pods --selector=name=weave-net -n kube-system --ignore-not-found=true
```

Comment:
Would it be possible to find the pod on this node in particular and delete that one?

`kubectl get pods -o wide -n kube-system -l name=weave-net | grep $(hostname)` or similar

Reply:
Are you suggesting it because of the `--ignore-not-found=true` usage? 
If so, I just it in case the pod (not created yet) be not found we still have a chance by re-trying to validate the network again. However, we could remove it. It has not a real reason to be there.

Reply:
No, I'm saying it because you're deleting all of the weave pods on all of the nodes when only the current node has a problem.

Reply:
Hun, by default this scenario should not occur and we will not delete the pods at all. If we check that the binaries in some edge case is not present I think the best approach is to delete the pods in all nodes since it can ensure that they will re-init, as we can check that adds the binaries in the required path and the config should be ok.

Reply:
I would disagree - I think we should only delete it for the current node. This way, we won't break a currently-running cluster - only one node at a time.